### PR TITLE
Adding inference to Linear Regression model

### PR DESCRIPTION
**Pull request by kavyasrinet**
Adding inference to the fluid `fit_a_line` model, this will help us when we re-write the models for the book.
It will also help @kexinzhao and me understand how inference works, before we start working on inference wrapper for Mobile.

We have a few questions:
1. Is the way we perform inference in this example correct ?
2. How do we determine the program that gets run inside of `exe.run` during inference ? (Should we clone the `fluid.default_main_program()` before we start training or is the method I used here, correct ? )

### CODE REVIEW

File: python/paddle/v2/fluid/tests/book/test_fit_a_line.py [hunk 1]
```diff
@@ -14,28 +16,46 @@
...
+                return  # if avg cost less than 10.0, we think our code is good.
+
+
+print("Training...")
+train()
+
+print("Now performing inference...")
+fluid.io.load_persistables(exe, "./fit_a_line.model/")
+for data in test_reader():
+    out, y_pred, y_label = exe.run(fluid.default_main_program(),
```
Comment (dzhwinter) [inline python/paddle/v2/fluid/tests/book/test_fit_a_line.py]: You must clone the main_program first.

Comment (QiJune) [reply python/paddle/v2/fluid/tests/book/test_fit_a_line.py]: Please refer to https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/fluid/tests/book/test_recognize_digits_mlp.py#L35.
We need to create a inference program first.

**Comment by dzhwinter** [comment]
> How do we determine the program that gets run inside of exe.run during inference? (Should we clone the fluid.default_main_program() before we start training or is the method I used here, correct ? )

```

inference_program = fluid.default_main_program().clone()
```

After discussed with @reyoung @QiJune , we believe that we should provide a general interface of train/test. Which means for most users, the framework will do the clone inside.

**Comment by kavyasrinet** [comment]
Thanks for pointing that out @dzhwinter and @QiJune . 
I have added an `inference_program` for the inference part as of now, and that is just a clone of the `main_program` since in this example, I am not fetching the accuracy or other attributes. Is my understanding correct here ?

**Pull request closed without merging**